### PR TITLE
server: make `host` configurable to enable IPv6 support

### LINKED ISSUE #359: ipv6 bind support
hello!

I've been testing this and it works great for systems on our internal network, which uses ipv4.

When we try connect from outside via ipv6 though, it's refused. It's not a firewall issue, IPs are already whitelisted.
It looks to only be binding on ipv4 addresses:

```
netstat -tulpn | grep lldap
tcp        0      0 0.0.0.0:3890            0.0.0.0:*               LISTEN      107866/lldap        
tcp        0      0 0.0.0.0:17170           0.0.0.0:*               LISTEN      107866/lldap        
```

I'm not seeing a config option for which interface to bind to either, so presumably it's not implemented?

Would it be possible to get ipv6 support? We can continue without for the affected machines but it'd be really sweet to get everything working with LLDAP :)

Thanks so much for your work, it's an awesome project!

### PR DESCRIPTION
**Pull request by wheinze**
Introduce a new configuration variable called `host` to enable IPv6 support.

If by any means, the naming of the variable should be adjusted, or a bool is preferred to toggle this behavior, I am for sure willing to adjust this ;)

Resolves #359

### CODE REVIEW

**Comment by nitnelave** [comment]
I think we should split the host into ldap and http parts (`api_host` and `ldap_host`): Some people will want to have LDAP only available on localhost for safety, whereas they will expose the API to the web.

**Comment by wheinze** [comment]
@nitnelave Thanks for the quick feedback and your hints on how to fully integrate into CLI as well ;)

I provided `ldap_host` and `api_host` as suggested by you.

I'm just wondering if it would make sense to align on other HTTP API vars and rather call it `http_host` :thinking: 
What do you think?

**Comment by nitnelave** [comment]
Oh yeah, definitely. I didn't have a look at the code when suggesting it, keeping it consistent is better

**Comment by wheinze** [comment]
> Oh yeah, definitely. I didn't have a look at the code when suggesting it, keeping it consistent is better

Adjusted to use `http_host` ;)

File: lldap_config.docker_template.toml [hunk 1]
```diff
@@ -7,9 +7,17 @@
 ## You can set it with the LLDAP_VERBOSE environment variable.
 # verbose=false
 
+## The host address that the LDAP server will be bound to.
+## To enable IPv6 support, simply switch "ldap_host" to "::1":
```
Comment (nitnelave) [inline lldap_config.docker_template.toml:11]: Can you add something like:

"To only allow connections from localhost (if you want to restrict to local self-hosted services), change it to "127.0.0.1" ("::1" in IPv6)"

And yes, btw, "::1" is localhost, you want to suggest "::" for the general case for IPv6.

**Review by nitnelave — COMMENTED** [review]
I take it you've tested it with IPv6? :)

Looks good, just a small comment on the config template.

File: lldap_config.docker_template.toml (hunk 1 above)
Comment (wheinze) [reply lldap_config.docker_template.toml]: > Can you add something like:
> 
> "To only allow connections from localhost (if you want to restrict to local self-hosted services), change it to "127.0.0.1" ("::1" in IPv6)"

Done: https://github.com/nitnelave/lldap/pull/377/commits/578d5e1fdb5c407f33c2d46f04519f20dd74c22d

> And yes, btw, "::1" is localhost, you want to suggest "::" for the general case for IPv6.

yep, sure thx. did correct it.

**Comment by wheinze** [comment]
> I take it you've tested it with IPv6? :)

yup, I tested all combinations with IPv4 and IPv6 for `ldap_host` and  `http_host` on my local dev setup.

**Pull request merged**